reconcile/utils/terrascript_aws_client.py

```diff
@@ -0,0 +1,4377 @@
+                            + f"unsupported region: {rule_region}"
+                        )
+                        continue
+                    values = {
+                        "type": "ingress",
+                        "from_port": 0,
+                        "to_port": 0,
+                        "protocol": "all",
+                        "cidr_blocks": [rule["cidr_block"]],
+                        "security_group_id": rule["security_group_id"],
+                    }
+                    if self._multiregion_account(req_account_name):
+                        values["provider"] = "aws." + rule_region
+                    rule_identifier = f"{identifier}-{rule['vpc_id']}"
+                    tf_resource = aws_security_group_rule(rule_identifier, **values)
+                    self.add_resource(req_account_name, tf_resource)
+
+    @staticmethod
+    def get_az_unique_subnet_ids(subnets_id_az):
+        """returns a list of subnet ids which are unique per az"""
+        results = []
+        azs = []
+        for subnet_id_az in subnets_id_az:
+            az = subnet_id_az["az"]
+            if az in azs:
+                continue
+            results.append(subnet_id_az["id"])
+            azs.append(az)
+
+        return results
+
+    def populate_resources(
+        self, namespaces, existing_secrets, account_name, ocm_map=None
+    ):
+        self.init_populate_specs(namespaces, account_name)
+        for specs in self.account_resources.values():
+            for spec in specs:
+                self.populate_tf_resources(spec, existing_secrets, ocm_map=ocm_map)
+
+    def init_populate_specs(self, namespaces, account_name):
+        self.account_resources = {}
+        for namespace_info in namespaces:
+            # Skip if namespace has no terraformResources
+            tf_provider_resources = (
+                namespace_info.get("terraformProviderResources") or []
+            )
+            tf_resources = namespace_info.get("terraformResources") or []
+            if not (tf_provider_resources or tf_resources):
+                continue
+            for provider_resource in tf_provider_resources:
+                provision_provider = provider_resource["provider"]
+                # Skip if provision provider is not aws
+                if provision_provider != "aws":
+                    continue
+                account = provider_resource["provisioner"]["name"]
+                # Skip if account_name is specified
+                if account_name and account != account_name:
+                    continue
+                if account not in self.account_resources:
+                    self.account_resources[account] = []
+                for resource in provider_resource["resources"]:
+                    resource["account"] = account
+                    populate_spec = {
+                        "resource": resource,
+                        "provision_provider": provision_provider,
```

Comment:
why is provision_provider stored in the populate_spec? it is always "aws" because non-aws resources are rightfully discarded in this context.

do i miss a place it is used?

Reply:
future compatibility with additional providers?